src/network/postgres/postgres_protocol_interpreter.cpp

```diff
@@ -83,7 +83,17 @@ Transition PostgresProtocolInterpreter::ProcessStartup(const common::ManagedPoin
       NETWORK_LOG_TRACE(db_name);
     }
   }
+
   auto oids = t_cop->CreateTempNamespace(context->GetConnectionID(), db_name);
+
+  uint32_t sleep_time = INITIAL_BACKOFF_TIME;
+  while ((oids.first == catalog::INVALID_DATABASE_OID || oids.second == catalog::INVALID_NAMESPACE_OID) &&
```

Comment:
Add documentation that explains what this loop is protecting from.

Reply:
It shouldn't sit in a loop with an `or` on these conditions. The former is an invalid database name provided. We don't want to spin in that case. We should only retry in the latter case (failed to create namespace). You'll see a split this logic in Extended Query PR already, but I'll just resolve the changes since yours will be merged first:

https://github.com/cmu-db/terrier/blob/65034c5b22ab87d29dbf889af0635fbe4edfeb38/src/network/postgres/postgres_protocol_interpreter.cpp

Also those returning error messages don't really work since Terminating the connection prevent any packets from being sent back to the client. We should eventually fix that.